tensorboard/webapp/runs/views/runs_table/runs_table_test.ts

```diff
@@ -691,6 +708,90 @@ describe('runs_table', () => {
         expect(saveButton!.textContent).toContain('Save');
       });
 
+      it('dispatches `runGroupByChanged` when regex editing is saved', () => {
+        store.overrideSelector(getEnabledColorGroup, true);
+        store.overrideSelector(getEnabledColorGroupByRegex, true);
+        store.overrideSelector(getRunGroupBy, {key: GroupByKey.EXPERIMENT});
+        const fixture = createComponent(
+          ['book'],
+          [RunsTableColumn.RUN_NAME, RunsTableColumn.RUN_COLOR]
+        );
+
+        const menuButton = fixture.debugElement
+          .query(By.directive(RunsGroupMenuButtonContainer))
+          .query(By.css('button'));
+        menuButton.nativeElement.click();
+
+        getColorGroupByHTMLElement(REGEXEDIT.REGEXEDIT)!.click();
+        const dialogContainer = overlayContainer
+          .getContainerElement()
+          .querySelector('mat-dialog-container');
+        const [cancelButton, saveButton] = dialogContainer!.querySelectorAll(
+          'button'
+        );
+
+        saveButton.click();
+        expect(dispatchSpy).toHaveBeenCalledWith(
+          runGroupByChanged({
+            experimentIds: ['book'],
+            groupBy: {key: GroupByKey.REGEX, regexString: ''},
+          })
+        );
+
+        const dialogInputDebugElement: DebugElement = new DebugElement(
+          overlayContainer
+            .getContainerElement()
+            .querySelector('mat-dialog-container input')
+        );
+        sendKeys(fixture, dialogInputDebugElement, 'foo(\\d+)');
+
+        saveButton.click();
+        expect(dispatchSpy).toHaveBeenCalledWith(
+          runGroupByChanged({
+            experimentIds: ['book'],
+            groupBy: {key: GroupByKey.REGEX, regexString: 'foo(\\d+)'},
+          })
+        );
+      });
+
+      it('does not dispatch `runGroupByChanged` with GroupByKey.REGEX when regex editing is cancelled', () => {
+        store.overrideSelector(getEnabledColorGroup, true);
+        store.overrideSelector(getEnabledColorGroupByRegex, true);
+        store.overrideSelector(getRunGroupBy, {key: GroupByKey.EXPERIMENT});
+        const fixture = createComponent(
+          ['book'],
+          [RunsTableColumn.RUN_NAME, RunsTableColumn.RUN_COLOR]
+        );
+
+        const menuButton = fixture.debugElement
+          .query(By.directive(RunsGroupMenuButtonContainer))
+          .query(By.css('button'));
+        menuButton.nativeElement.click();
```

Comment:
No action required: well, I wanted this part to be within the utility function but it is fine as is.